erpnext/public/js/controllers/transaction.js

```diff
@@ -1264,6 +1267,43 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 				}
 			}
 		});
+
+		const transaction_pricing_rule = this.frm.call({
+			method: "erpnext.accounts.doctype.pricing_rule.pricing_rule.apply_transaction_pricing_rule",
+			args: {	doc: me.frm.doc },
+			callback: function(r) {
+				if (!r.exc && r.message) {
+					r = r.message;
+					if ('price' in r) {
```

Comment:
why not checking the value of the field price_or_product_discount

Reply:
Return doesn't have `apply_transaction_pricing_rule` field.